packages/upload-client/src/index.js

```diff
@@ -129,7 +130,27 @@ async function uploadBlockStream(conf, blocks, options = {}) {
           const multihashDigest = await hasher.digest(bytes)
           /** @type {import('@web3-storage/capabilities/types').PieceLink} */
           const piece = Link.create(raw.code, multihashDigest)
+
+          // Invoke store/add and write bytes to write target
           const cid = await Store.add(conf, bytes, options)
+          // Invoke filecoin/offer for data
+          const result = await Storefront.filecoinOffer(
+            {
+              issuer: conf.issuer,
+              audience: conf.audience,
+              // Resource of storefront is the issuer did
+              with: conf.issuer.did(),
+              proofs: conf.proofs,
+            },
+            cid,
+            piece,
+            options
+          )
+
+          if (result.out.error) {
+            throw result.out.error
```

Comment:
I think this is consistent with what we're doing elsewhere in the client:

```suggestion
            throw new Error('failed to offer piece for aggregation into filecoin deal', { cause: result.out.error })
```